pkg/obfuscate/sql_test.go

```diff
@@ -93,6 +93,41 @@ func TestDollarQuotedFunc(t *testing.T) {
 	})
 }
 
+func TestSingleDollarIdentifier(t *testing.T) {
+	q := `
+	MERGE [md].InventoryLotLocationView as target
+	using (select *
+	from #InventoryLotLocationView_Loader) as source
+		on (target.[LotId] = source.[LotId]
+		and target.[ItemKey] = source.[ItemKey]
+		and target.[LocationCode] = source.[LocationCode])
+	when matched 
+	and  (target.[LotNumber] <> source.[LotNumber]
+		or target.[ExpirationDate] <> source.[ExpirationDate]
+	)
+	then update
+	set target.[LotNumber] = source.[LotNumber] 
+	when not matched by source then delete
+	when not matched by target then
+	insert([LotId] 
+	,[ItemKey] 
+	)
+	values (
+	source.[LotId] 
+	,source.[ItemKey] 
```

Comment:
Can we simplify that query and still have a test case? It might be nicer to have a minimal query that reproduces the scenario even if it looks less realistic.

Reply:
I've made a simplified test query. although the updated version is still quite complex because the specific test case requires a form of `MERGE .... WHEN MATCHED UPDATE ... WHEN NOT MATCHED DELETE ... OUTPUT ...`

Reply:
Ok yes, this makes sense. Thanks for doing this. 🙇